mailbag/controller.py

```diff
@@ -56,14 +59,20 @@ def generate_mailbag(self):
         csv_portion = []
         csv_portion.append(header)
 
-
+        port = 5000
+        if not self.args.dry_run:
+            _thread.start_new_thread(helper.startServer,())
+            
         for message in mail_account.messages():
             # do stuff you ought to do per message here
 
             # Generate mailbag_message_id
             mailbag_message_id += 1
             message.Mailbag_Message_ID = mailbag_message_id
 
+            if message.HTML_Body:
+                helper.saveWARC(self.args.dry_run,warc_dir,message,port)
```

Comment:
This should be run from a warc.py file in `derivatives` just like the other derivatives. Is there a reason why this is in the controller? At the very least, it needs to be compartmentalized a lot better.